src/core/MaverickV2.sol

```diff
@@ -248,27 +315,14 @@ abstract contract MaverickV2 is SettlerAbstract {
         }
         if (sellAmount == 0) {
             sellAmount = sellToken.fastBalanceOf(address(pool));
-            IMaverickV2Pool.State memory poolState = pool.getState();
             unchecked {
-                sellAmount -= tokenAIn ? poolState.reserveA : poolState.reserveB;
+                sellAmount -= pool.fastGetReserveAOrB(tokenAIn);
```

Comment:
I think this logic is now backwards. By using `tokenAIn.toUint() << 5` as the offset, you read `reserveB` when `tokenAIn` is true and `reserveA` when false.

Reply:
Good catch, weird this was not detected by the tests.

Reply:
That branch of the code is not covered by the test, adding a test for it ...